torch/ao/sparsity/_experimental/data_sparsifier/lightning/callbacks/_data_sparstity_utils.py

```diff
@@ -0,0 +1,36 @@
+import logging
+from torch.ao.sparsity._experimental.data_sparsifier.base_data_sparsifier import SUPPORTED_TYPES
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _attach_model_to_data_sparsifier(module, data_sparsifier):
+    """Attaches a data sparsifier to all the layers of the module.
+    Essentialy, loop over all the weight parameters in the module and
+    attach it to the data sparsifier.
+    Note::
+        The '.' in the layer names are replaced with some other character
+        before attaching to the sparsifier. This is because, the data
+        sparsifier uses a dummy model inside to store the weight parameters.
```

Comment:
Be more specific: what do you mean "with some other character"?

Reply:
Also, I don't think a layer name can have a `.` in there. Do you support FQNs?

Reply:
Yes, the layer name cannot have a ```.```. We're basically taking the name of the model parameter and replacing it with ```_```.